inference-engine/src/multi_device/multi_device.cpp

```diff
@@ -525,24 +526,22 @@ void MultiDeviceInferencePlugin::QueryNetwork(const ICNNNetwork&
     }
 
     DeviceMap<DeviceInformation> metaDevices = ParseMetaDevices(priorities->second, fullConfig);
-    std::map<std::string, QueryNetworkResult> queryResults;
-
+    std::unordered_set<std::string> supportedLayers;
     for (auto&& value : metaDevices) {
         auto& deviceName = value.first;
         auto& metaDevice = value.second;
-        queryResults[deviceName] = GetCore()->QueryNetwork(network, deviceName, metaDevice.config);
-    }
-
-    details::CNNNetworkIterator i(&network);
-    while (i != details::CNNNetworkIterator()) {
-        CNNLayer::Ptr layer = *i;
-        bool layerIsInQueryResultsForAllDevices = std::all_of(std::begin(queryResults), std::end(queryResults),
-            [&](const std::map<std::string, QueryNetworkResult>::value_type& qr) {
-                return qr.second.supportedLayersMap.end() != qr.second.supportedLayersMap.find(layer->name);});
-        if (layerIsInQueryResultsForAllDevices) {
-            queryResult.supportedLayersMap[layer->name] = GetName();
+        auto clonedNetwork = cloneNetwork(network);
+        auto deviceQr = GetCore()->QueryNetwork(*clonedNetwork, deviceName, metaDevice.config);
```

Comment:
What happens if one plugin works with nGraph representation and another with legacy format?

Reply:
Added conversion to legacy implemenation if plugin without ngraph support found